### PR TITLE
configure: minor cleanup

### PR DESCRIPTION
**Pull request by garlick**
This modernizes some of the autoconf macros used by the project and some other related tidying up.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -5,7 +5,6 @@ SUBDIRS = \
 	libdiod \
 	diod \
 	utils \
-	scripts \
```
Comment (grondo) [inline Makefile.am:8]: Commit message nit: "Makefile.am's". Suggestion: Makefile.ams or  'Makefile.am files'

File: autogen.sh [hunk 2]
```diff
@@ -1,17 +1,4 @@
...
-automake --copy --add-missing 
-echo "Running autoconf ... "
-autoconf
-echo "Cleaning up ..."
-mv aclocal.m4 config/
-rm -rf autom4te.cache
-echo "Now run ./configure to configure diod for your environment."
-
+echo "Running autoreconf --force --verbose --install"
+autoreconf --force --verbose --install || exit
```
Comment (grondo) [inline autogen.sh:3]: Commit message: s/autoregen/autoreconf/

**Review by grondo — APPROVED** [review]
LGTM! Just noticed a couple trivial commit message nits

**Comment by garlick** [comment]
Thanks!  Fixed that and I'll set MWP.

**Pull request merged**